docs/userguide/datafiles.rst

```diff
@@ -2,12 +2,20 @@
 Data Files Support
 ====================
 
-Old packaging installation methods in the Python ecosystem have
-traditionally allowed the inclusion of "data files" (files beyond
-:ref:`the default set <manifest>` ), which are placed in a platform-specific
-location. However, the most common use case for data files distributed
-with a package is for use *by* the package, usually by including the
-data files **inside the package directory**.
+In the Python ecosystem, the term "data files" is used in various complex scenarios
+and can have nuanced meanings.
+For the purposes of this documentation, we define "data files" as non-Python files
+that are installed alongside Python modules and packages on the user's machine
+when they install a :term:`distribution <Distribution Package>` from PyPI
+or via a ``.whl`` file.
```

Comment:
I would tag and explain something I want to change (I find myself lose track of what I was originally intended if I overthink), please free feel to comment.

I believe the source (PyPI vs a `.whl` file) is not important here, but whether we're installing from a binary distribution (`.whl`) or source distribution that matters.

Reply:
We are probably better off if we don't define data files in terms of source distributions or source trees...

Reply:
> We are probably better off if we don't define data files in terms of source distributions or source trees

Sorry it looks like I might have caused confusion here. I was suggesting:
```diff
- When they install a distribution from PyPI or via a `.whl` file
+ When they install a distribution from source distribution or via a `.whl` file
```

Because `install from PyPI` is not important here, because by `install from PyPI`, we could install both from `.whl` file (if wheel is built for that particular platform) or source distribution. I hope I didn't misunderstand anything?

Reply:
I think the former phrase is better for the definition than the second. Alternatively only the following is also fine:

```diff
- When they install a distribution from source distribution or via a `.whl` file
+ when they install a distribution via a `.whl` file
```

I believe We should not define define data files as per `source distribution`. A Python package always need to be transformed into a wheel, so the files inside of the wheel is what actually matters.

In the previous version `install from PyPI` is being used as an equivalent of `pip install package`, which in the end of the day downloads (preferentially) a `.whl` file.